lib/dao.js

```diff
@@ -2426,7 +2470,19 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
     inst.setAttributes(data);
 
     inst.isValid(function (valid) {
-      if (!valid) {
+
```

Comment:
> But then some tests concerning the lifecycle hooks failed, because the beforeValidation and afterValidation hooks where missing. 

Hmm, that's interesting and and rather unexpected. Could you please provide more details on the test failures? I am surprised that an existing test fails when this change should be fully backwards compatible.